src/execution.jl

```diff
@@ -704,9 +708,17 @@ macro bprofile(args...)
     return esc(
         quote
             local $tmp = $BenchmarkTools.@benchmarkable $(args...)
-            $BenchmarkTools.warmup($tmp)
-            $(hasevals(params) ? :() : :($BenchmarkTools.tune!($tmp)))
+            $(
+                if hasevals(params)
+                    :($tmp.samplefunc(
+                        $tmp.quote_vals, $BenchmarkTools.Parameters($tmp.params; evals=1)
+                    ))
+                else
+                    :($BenchmarkTools.tune!($tmp))
+                end
+            )
```

Comment:
Why is this only needed in `@bprofile`?

Reply:
This was a bit of a compromise. `run` used to skip warmups. This PR changed it so that `run` now warms up. Unfortunately, `@bprofile` profiles `run`, and is now profiling the warmup run (potentially profiling compilation costs). My quick fix was to warm up before calling `run`, so `@bprofile` profiles a redundant warmup.

A more robust fix would be to give `run` the option to skip the warm up. This was suggested by @zentrik. This is a bigger change, but would not require `@bprofile` to change semantics.